src/main/java/seedu/weme/ui/CommandBox.java

```diff
@@ -40,9 +48,74 @@ public CommandBox(CommandExecutor commandExecutor, CommandPrompter commandPrompt
                 isShowingCommandSuccess = false;
             }
         });
-        commandTextField.addEventFilter(KeyEvent.KEY_PRESSED, this::handleTabPress);
+        commandTextField.addEventFilter(KeyEvent.KEY_PRESSED, this::handleKeyPress);
+    }
+
+    /**
+     * Handles the key press event, {@code keyEvent}.
+     */
+    @FXML
+    private void handleKeyPress(KeyEvent keyEvent) {
+        switch (keyEvent.getCode()) {
+        case UP:
+            // As up and down buttons will alter the position of the caret,
+            // consuming it causes the caret's position to remain unchanged
+        case DOWN:
+            handleLikeByKeyPress(keyEvent);
+            break;
+        case LEFT:
+        case RIGHT:
+            handleIndexToggleByKeyPress(keyEvent);
+            break;
+        case TAB:
+            handleTabPress(keyEvent);
+            break;
+        default:
+            // let JavaFx handle the keypress
+        }
+    }
+
+    /**
+     * Changes index in the command box with key press to enable fast liking.
+     */
+    private void handleIndexToggleByKeyPress(KeyEvent event) {
+        final Matcher matcher = BASIC_COMMAND_FORMAT.matcher(commandTextField.getText().trim());
+        if (matcher.matches()) {
+            final String commandWord = matcher.group(WemeParser.COMMAND_WORD);
+            final String argument = matcher.group(WemeParser.ARGUMENTS);
+            int change = event.getCode().equals(KeyCode.LEFT) ? -1 : 1;
+            int currentLikeCount = Integer.parseInt(argument.trim());
+            int newLikeCount = currentLikeCount + change;
+            commandTextField.setText(commandWord + " " + (newLikeCount < 1 ? 1 : newLikeCount));
+        }
+        commandTextField.positionCaret(commandTextField.getText().length());
```

Comment:
we have a problem, now, whatever command you put, you can't shift the caret left or right via the left and right arrow keys, fix and can merge

Reply:
oh it's captured. I need to move the match out in the handleKeyPress.